apps/voting-disputable/contracts/DisputableVoting.sol

```diff
@@ -543,13 +470,76 @@ contract DisputableVoting is DisputableApp {
     }
 
     /**
-    * @dev Void an entry
+    * @dev Void a vote
     * @param _voteId Identification number of the entry to be voided
     */
     function _onDisputableActionVoided(uint256 _voteId) internal {
+        // When a vote that has been challenged gets voided, it is considered allowed for the current implementation.
+        // This could be the case for challenges where the arbitrator refuses to rule.
         _onDisputableActionAllowed(_voteId);
     }
 
+    // Internal fns
+
+    /**
+    * @dev Internal function to create a new vote
+    * @return voteId id for newly created vote
+    */
+    function _newVote(bytes _executionScript, string _metadata) internal returns (uint256 voteId) {
+        uint64 snapshotBlock = getBlockNumber64() - 1; // avoid double voting in this very block
+        uint256 votingPower = token.totalSupplyAt(snapshotBlock);
+        require(votingPower > 0, ERROR_NO_VOTING_POWER);
+
+        voteId = votesLength++;
+
+        Vote storage vote_ = votes[voteId];
+        vote_.startDate = getTimestamp64();
+        vote_.snapshotBlock = snapshotBlock;
+        vote_.overruleWindow = overruleWindow;
+        vote_.supportRequiredPct = supportRequiredPct;
+        vote_.minAcceptQuorumPct = minAcceptQuorumPct;
+        vote_.votingPower = votingPower;
+        vote_.executionScript = _executionScript;
+        vote_.disputableStatus = DisputableStatus.Active;
+
+        // Notify the Agreement app tied to the current voting app about the vote created.
+        // This is mandatory to make votes disputable, we then store its reference on the Agreement app.
+        vote_.actionId = _newAgreementAction(voteId, msg.sender, bytes(_metadata));
+
+        emit StartVote(voteId, msg.sender, _metadata);
+    }
+
+    /**
+    * @dev Internal function to cast a vote. It assumes the queried vote exists.
```

Comment:
If we have this assumption, I wonder if we should prefix with `unsafe`?

Or perhaps better, is just to use the `Vote storage` variable.

Reply:
We are not using a storage pointer cause we rely on the vote ID for events :/
I'm a bit hesitant about adding the `unsafe` prefix to half of the internal methods haha

Reply:
But we won't be able to emit the events then, or do you mean to receive both a `uint256 voteId` and `Voting storage vote` reference?